src/lib/types/Challenge.ts

```diff
@@ -24,8 +24,12 @@ export type Challenge = Timestamps &
 			type: 'progress';
 			at: Date;
 			order: Order['_id'];
+			amountPerProduct?: Record<string, number>;
```

Comment:
Oh I meant for this to be in the global zone, near  `ratio` and `globalRatio`

This is why I suggested using `$inc`